app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,37 @@
 class PostsController < ApplicationController
-  def index; end
+  def index
+    @user = User.find(params[:user_id])
+    @posts = @user.posts
+  end
 
-  def show; end
+  def show
+    @user = User.find(params[:user_id])
+    @post = @user.posts.find(params[:id])
+    @comments = @post.comments
+    @likes = @post.likes
+  end
+
+  def new
+    @current_user = current_user
+    @new_post = Post.new
+  end
+
+  def create
+    @new_post = Post.new(post_params)
+    @new_post.user = current_user
+
+    if @new_post.save
+      @new_post.update_post_counter
+      flash[:notice] = 'You have successfully created a new post.'
+      redirect_to user_path(@new_post.user_id), notice: 'Post created successfully'
+    else
+      render :new, alert: 'An error occurred while creating the post'
+    end
+  end
```

Comment:
- [ ] Please kindly take note that this implementation to create a new post is not yet effective because you did not config routes for `post#new and post#create`.

I'm suggesting this 👇 to be updated in your `config/routes.rb` file. **This has catered for comment#new and comment#create, likes#create routes as well**

```
resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show, :new, :create] do
    resources :likes, only: [:create]
      resources :comments, only: [:new, :create]
  end
```

**Note: I couldn't comment in the routes file because it's not part of changes files**